assets/app/scripts/directives/metrics.js

```diff
@@ -169,83 +203,99 @@ angular.module('openshiftConsole')
+
+            var usage = scope.usageByMetric[metricID] = {
+              total: getLimit(metricID)
+            };
+
+            var mostRecentValue = _.last(metricData).value;
+            if (isNaN(mostRecentValue)) {
+              mostRecentValue = 0;
+            }
+            if (/network\/rx|tx/.test(metricID) || metricID === 'memory/usage') {
+              mostRecentValue = bytesToMiB(mostRecentValue);
+            }
+
+            // Round to the closest whole number for the utilization chart.
+            usage.used = d3.round(mostRecentValue);
+            if (usage.total) {
+              usage.available = Math.max(usage.total - usage.used, 0);
+            }
 
-          angular.forEach(data, function(point) {
-            dates.push(point.timestamp);
-            if (point.value === undefined || point.value === null) {
-              // Don't attempt to round null values. These appear as gaps in the chart.
-              values.push(point.value);
-            } else {
-              switch (metric.id) {
-                case 'memory/usage':
-                  values.push(d3.round(bytesToMiB(point.value), 2));
-                break;
-                case 'cpu/usage':
-                  values.push(d3.round(point.value));
-                break;
+            angular.forEach(metricData, function(point) {
+              dates.push(point.timestamp);
+              if (point.value === undefined || point.value === null) {
+                // Don't attempt to round null values. These appear as gaps in the chart.
+                values[metricID].push(point.value);
+              } else {
+                switch (metricID) {
+                  case 'memory/usage':
+                  case 'network/rx':
+                  case 'network/tx':
+                    values[metricID].push(d3.round(bytesToMiB(point.value), 2));
+                    break;
+                  default:
+                    values[metricID].push(d3.round(point.value));
+                }
+              }
+            });
+
+            // Donut
+            var donutConfig, donutData;
+            if (usage.total) {
+              donutData = {
+                type: 'donut',
+                columns: [
+                  ['Used', usage.used],
+                  ['Available', usage.available]
+                ],
+                colors: {
+                  Used: "#0088ce",      // Blue
+                  Available: "#d1d1d1"  // Gray
+                }
+              };
+
+              if (!donutByMetric[metricID]) {
+                donutConfig = createDonutConfig(metric);
+                donutConfig.data = donutData;
+                $timeout(function() {
+                  donutByMetric[metricID] = c3.generate(donutConfig);
+                });
+              } else {
+                donutByMetric[metricID].load(donutData);
               }
             }
           });
 
+          var columns = _([dates]).concat(_.values(values)).value();
```

Comment:
Unless I'm doing something wrong, I think `_.concat` (as a function) is not available in the version of lodash we are using (only `concat` in the `_` prototype).